### PR TITLE
Lambdaworks wrapper

### PR DESCRIPTION
**Pull request by toni-calvin**

### CODE REVIEW

**Review by mmsc2 — APPROVED** [review]

**Review by fmoletta — COMMENTED** [review]
Looks Good!
I think we should add a `FromHex` method, as with this api we can only create a Felt from uint64

File: pkg/lambdaworks/lambdaworks.go [hunk 1]
```diff
@@ -1,12 +1,135 @@
...
+// Writes the result variable with the sum of a and the elements in b array.
+func (a Felt) AddFelts(felts []Felt) Felt {
+	var a_c C.felt_t = a.toC()
+	for _, felt := range felts {
+		var f_c C.felt_t = felt.toC()
+		C.add(&a_c[0], &f_c[0], &a_c[0])
+	}
+
+	return fromC(a_c)
+}
```
Comment (fmoletta) [inline pkg/lambdaworks/lambdaworks.go]: We don't have a use-case for this, nor `SubFelts`

File: pkg/lambdaworks/lambdaworks.go [hunk 2]
```diff
@@ -1,12 +1,127 @@
...
+	return fromC(result)
+}
+
+func FromHex(value string) (Felt, error) {
+	val, err := hexToUint64(value)
+	if err != nil {
+		return From(0), err
+	}
+	return From(val), nil
+}
```
[the remark this answers is not in the document]
Comment (toni-calvin) [reply pkg/lambdaworks/lambdaworks.go]: Solved! Thanks for your help

File: pkg/lambdaworks/lambdaworks.go (hunk 1 above)
Comment (toni-calvin) [reply pkg/lambdaworks/lambdaworks.go]: Corrected!

File: pkg/lambdaworks/lambdaworks.go [hunk 3]
```diff
@@ -1,12 +1,113 @@
...
+func fromC(result C.felt_t) Felt {
+	var limbs [4]Limb
+	for i, limb := range result {
+		limbs[i] = Limb(limb)
+	}
+	return Felt{limbs: limbs}
+}
+
+// Gets a Felt representing the "value" number, in Montgomery format.
+func From(value uint64) Felt {
```
Comment (fmoletta) [inline pkg/lambdaworks/lambdaworks.go]:
```suggestion
func FeltFromUint64(value uint64) Felt {
```
Lets make this name more descriptive

**Review by fmoletta — APPROVED** [review]
Looks good 🚀 , left small suggestion

**Pull request merged**